src/vs/workbench/api/common/extHostTerminalService.ts

```diff
@@ -809,8 +812,12 @@ export abstract class BaseExtHostTerminalService extends Disposable implements I
 	}
 
 	public $acceptDefaultProfile(profile: ITerminalProfile, automationProfile: ITerminalProfile): void {
+		const oldProfile = this._defaultProfile;
 		this._defaultProfile = profile;
 		this._defaultAutomationProfile = automationProfile;
+		if (!oldProfile || oldProfile.path !== profile.path) {
+			this._onDidChangeShell.fire(profile.path);
+		}
```

Comment:
I think this would simplify things:

```suggestion
		this._defaultProfile = profile;
		this._defaultAutomationProfile = automationProfile;
		if (oldProfile?.path !== profile.path) {
			this._onDidChangeShell.fire(profile.path);
		}
```